### PR TITLE
Maint-26792: Created and renamed space with special character accepted

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**
Problem:  Renamed space with special character accepted so  I added a control in the DisplayName field.

### CODE REVIEW

File: webapp/portlet/src/main/webapp/space-settings/components/SpaceSettingFormDrawer.vue [hunk 1]
```diff
@@ -214,14 +223,20 @@ export default {
       }
     },
     saveSpace(event) {
-      if (event) {
-        event.preventDefault();
-        event.stopPropagation();
+      const regex = /^[A-Za-z0-9& ]+$/;
```
Comment (ahamdi) [inline webapp/portlet/src/main/webapp/space-settings/components/SpaceSettingFormDrawer.vue]: Please use the same pattern as in org.exoplatform.social.core.space.SpaceUtils#isValidSpaceName

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/webapp/spaces-list/components/ExoSpaceFormDrawer.vue [hunk 2]
```diff
@@ -1,198 +1,206 @@
...
+              {{ $t('spacesList.button.updateSpace') }}
+            </template>
+            <template v-else>
+              {{ $t('spacesList.button.createSpace') }}
+            </template>
+          </v-btn>
+        </div>
+      </template>
+    </exo-drawer>
+    <exo-confirm-dialog
```
Comment (ahamdi) [inline webapp/portlet/src/main/webapp/spaces-list/components/ExoSpaceFormDrawer.vue]: Please remove the confirm popup and use the inline alert message

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/webapp/space-settings/components/SpaceSettingFormDrawer.vue [hunk 3]
```diff
@@ -1,135 +1,143 @@
 <template>
-  <exo-drawer
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/space-settings/components/SpaceSettingFormDrawer.vue]: can you please keep the format of the file to make the PR readable, thanks.

Comment (Rdidmohamedelhadi) [reply webapp/portlet/src/main/webapp/space-settings/components/SpaceSettingFormDrawer.vue]: ok

File: component/service/src/main/java/org/exoplatform/social/rest/impl/space/SpaceRestResourcesV1.java [hunk 4]
```diff
@@ -442,8 +442,8 @@ public Response updateSpaceById(@Context UriInfo uriInfo,
     if (space == null || (! spaceService.isManager(space, authenticatedUser) && ! spaceService.isSuperManager(authenticatedUser))) {
       throw new WebApplicationException(Response.Status.UNAUTHORIZED);
     }
-    if (space.getDisplayName().length() > 200 || !SpaceUtils.isValidSpaceName(space.getDisplayName())) {
-      throw new WebApplicationException(Response.Status.BAD_REQUEST);
+    if (model == null || model.getDisplayName() == null || model.getDisplayName().length() == 0 || model.getDisplayName().length() > 200 || !SpaceUtils.isValidSpaceName(model.getDisplayName())) {
```
Comment (ahamdi) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/space/SpaceRestResourcesV1.java]: model == null was already checked above
model.getDisplayName().length() == 0 could be replaced with model.getDisplayName().isEmpty() or better you can use StringUtils.isBlank(model.getDisplayName()) to replace both checks

**Review by ahamdi — CHANGES_REQUESTED** [review]

**Review by ahamdi — APPROVED** [review]
LGTM

**Pull request merged**